File: 通过coreGraphic把图片变灰/通过coreGraphic把图片变灰/ViewController.m
//
//  ViewController.m
//  通过coreGraphic把图片变灰
//
//  Created by 黄成都 on 15/7/4.
//  Copyright (c) 2015年 黄成都. All rights reserved.
//

#import "ViewController.h"
#import "UIImage+changeColor.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *originalImage;

@property (weak, nonatomic) IBOutlet UIImageView *grayImage;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.grayImage.image = [self.originalImage.image grayImage:self.originalImage.image];
    //self.grayImage.image = [self.originalImage.image imageWithColor:[UIColor redColor]];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: swift代码段/基础部分/基础部分/ViewController.swift
//
//  ViewController.swift
//  基础部分
//
//  Created by 黄成都 on 15/8/18.
//  Copyright (c) 2015年 黄成都. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        //元组
        let http404Error = (404,"Not Found")
        let (statusCode,statusMessage) = http404Error
        println("The status code is \(statusCode)")
        println("the status message is \(statusMessage)")
        println("the status code is \(http404Error.0)")
        println("=====================================")
        let http200Status = (statusCode1:200,description1:"ok")
        println("The status code is \(http200Status.statusCode1)")
        println("=============可选类型================")
        let possibleNumber = "123"
        let convertedNumber:Int? = possibleNumber.toInt()
        println(convertedNumber)
        println("=============隐式可选类型=========")
        let possibleString:String? = "An optional string."
        let forcedString:String = possibleString!
        let assumedString:String! = "An implicitly unwrapped."
        let implicitString:String = assumedString
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

